OpenTelemetry.DotNet.Auto.psm1

```diff
@@ -240,13 +240,25 @@ function Install-OpenTelemetryCore() {
         [System.Reflection.Assembly]::Load("System.EnterpriseServices, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a") | Out-Null
         $publish = New-Object System.EnterpriseServices.Internal.Publish 
         $dlls = Get-ChildItem -Path $installDir\netfx\ -Filter *.dll -File
-        for ($i = 0; $i -le $dlls.Count; $i++) {
+        for ($i = 0; $i -lt $dlls.Count; $i++) {
             $percentageComplete = $i / $dlls.Count * 100
             Write-Progress -Activity "Registering .NET Framweworks dlls in GAC" `
                 -Status "Module $($i+1) out of $($dlls.Count). Installing $($dlls[$i].Name):" `
                 -PercentComplete $percentageComplete
+
+            if ($dlls[$i].Name -eq "netstandard.dll") {
```

Comment:
maybe just to define ignore list and check dll name not in the list? prob scales better in the future.

Reply:
and we can have it in one place, instead of two

Reply:
Whenever a new assembly gets added we need to review the list.

Reply:
It looks like the explorations in #1984 pointed a good direction in using C# to determine whether a DLL is an assembly, and the blockers revolves around having to download/unload dependencies in PowerShell and having to do C# things in PowerShell.

I have an idea to automate it which combines the option 1 and option 2: if we generate the "ignore list" or "allow list" in test projects like in #1156 or build projects like in #2073, there is no need to worry about having to load/unload C# packages in PowerShell module.

The list can be generated in a text file or directly as a few lines of HashMap initialization in the `OpenTelemetry.DotNet.Auto.psm1` file itself to avoid having to read the text file from PowerShell, kind of like the auto-generated block "Transient dependencies auto-generated by GenerateNetFxTransientDependencies" in the `OpenTelemetry.AutoInstrumentation.csproj` file.

Reply:
I will address it in an "easy" way first. https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/2171#discussion_r1097978479 is a good idea for a separate PR. @xiang17 do you want to try it?

Reply:
Addressed https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/2171/commits/692d9cb1b73035575ae23c43ee524760589c0aba

Reply:
I got a few higher priority tasks recently. I might not have time for it until later.